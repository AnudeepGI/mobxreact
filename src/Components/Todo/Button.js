import React from 'react';

const Button = (props) => {
    console.log(props)
    return (
        <button onClick={props.action}>
            {props.title}
        </button>
    )
}

export default Button;