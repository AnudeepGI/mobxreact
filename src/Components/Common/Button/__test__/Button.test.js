import React from 'react';
import ReactDOM from 'react-dom';
import Button from '..';

import { render } from '@testing-library/react'
import '@testing-library/jest-dom'

import renderer from 'react-test-renderer';


it("rednder without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Button/>,div)
});

it("rednder button", () => {
    const title = "Click"
    const { container } = render(<Button title={title} />)
    const button = container.querySelector('button');
    expect(button).toHaveTextContent(title);
});

it("Match Snapshot - Button", () => {
    const tree = renderer.create(<Button title="Click"></Button>).toJSON();
    expect(tree).toMatchSnapshot();
});






