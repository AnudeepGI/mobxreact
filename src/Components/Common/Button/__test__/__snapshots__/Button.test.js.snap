// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match Snapshot - Button 1`] = `
<button>
  Click
</button>
`;

exports[`Match Snapshot 1 1`] = `
<button>
  Click
</button>
`;
